spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -307,6 +312,8 @@ class SpdxLicenseWithExceptionExpression(
 
     override fun simpleLicense() = license.toString()
 
+    override fun exception() = exception
```

Comment:
Would it make sense to make this `exceptions(): List<String>` for consistency with the below `licenses()`?

Reply:
The list variant does not make sense because an `SpdxSingleLicenseExpression` can never have more than one exception, and I don't want to use `simpleException` because unlike with licenses there are no complex exceptions.

Reply:
> The list variant does not make sense because an `SpdxSingleLicenseExpression` can never have more than one exception ...

I had in mind to place the list variant function on the same class hierarchy level as the list variant function for the licenses. Assuming that your above contra argument doesn't apply I believe. 

> .. , and I don't want to use `simpleException` because unlike with licenses there are no complex exceptions.

With complex I assume you are referring to `compound expressions`. If this is true then I don't see any difference between `exceptions` and licenses there as multiple exceptions can be used in `compound expressions` so also the exceptions can be operands to `and` and `or`. Does this make sense, or didn't I get your above point?

Reply:
The helper functions are for `SpdxSingleLicenseExpression` because this type is used in several places and the helper functions are used when working with single licenses, not complex expressions. So yes, if there is a need for it adding an `exceptions()` function to `SpdxExpression` would make sense, but right now I need exactly the function I added here.

IIRC the name `simpleLicense` was proposed by @sschuberth in a review, because this term is used in the grammar of SPDX expressions for expressions that contain only a single license in contrast to complex expressions that contain multiple licenses. However, there is no similar naming for exceptions, because exceptions are not SPDX expressions on their own, but always have to be concatenated to a license using `WITH`.

Reply:
Then let's keep it as is. If the name turns out to be misleading it can be improved later on easily.